cranelift/reader/src/run_command.rs

```diff
@@ -22,8 +22,48 @@ pub enum RunCommand {
     Run(Invocation, Comparison, Vec<DataValue>),
 }
 
+impl RunCommand {
+    /// Trim leading spaces and comment chars (i.e. ";") from a string.
+    pub fn trim_comment_chars(comment: &str) -> &str {
+        comment.trim_start_matches(|c| c == ' ' || c == ';')
+    }
+
+    /// Check if a CLIF comment is potentially parseable as a [RunCommand].
+    pub fn is_potential_run_command(comment: &str) -> bool {
+        let trimmed = Self::trim_comment_chars(comment);
+        trimmed.starts_with("run") || trimmed.starts_with("print")
```

Comment:
Yeah, this is a good direction. These weird `trimmed_comment_chars` and `is_potential_run_command` functions were trying to do some lookahead to see if we actually needed to really parse the comment. I moved that logic into `parse_run_command` (which is in the parser module, where this functionality should be) and made it return `ParseResult<Option<RunCommand>>` instead of `ParseResult<RunCommand>`. The signature is a bit more complex but I think it is headed more in the direction you are suggesting.